### ISSUE TITLE
Does orleans transfer grain module to another silo?

### ISSUE THREAD

**Issue by christallire** [dotnet/orleans#8730]
I have a strange situation.

I'm injecting singleton service to grain, when I rollout new version of grain the service is created. and then strange thing is happened.
I put a log to make sure the singleton service is created only once but it was created twice.
eventually renamed various methods to make sure it's really same code but somehow the old service is retrieved and injected. 
the symptom gone once I put the entire cluster down and restart again.

so my question is: Does orleans transfer grain module to another silo?
It doesn't make sense unless I get old module back.

.NET8 / orleans 7.2.3

**Comment by christallire** [comment]
If so, is there anyway to disable it?

**Comment by ReubenBond** [comment]
Orleans doesn't transfer dlls to other hosts - it does no magic with dlls at all. If you are seeing double logging of some thing, it's either a logging issue (eg logs are being duplicates, possibly misconfigured) or a dependency injection issue (eg multiple DI containers or incorrect registration)

**Comment by christallire** [comment]
That's odd. I tried to rename method with `[MethodImpl(MethodImplOptions.NoInlining)]` to avoid inlining and captured a few stacktraces when it happened. both methods (before and after renaming) exists... :|

Thanks anyways, I'll get back here if I find something

**Comment by ReubenBond** [comment]
If you show some code, etc, we might be able to help

**Comment by ReubenBond** [comment]
I'll close this now, since the initial question has been answered. Please open a new issue and reference this one if you do find something we can help with.
